### ISSUE TITLE
TIFFImageReader: Ignore bogus StripByteCounts entry with value 0

### ISSUE THREAD

**Issue by stbstb** [haraldk/TwelveMonkeys#449]
I have to read a CCITT-G4-compressed TIFF image created by a customer's scanner. The TIFF image contains 0 as the value for StripByteCounts. libtiff just ignores this obviously invalid value, TwelveMonkeys' TIFFImageReader reads the image as completely black.
If TIFFImageReader ignores this value (as if StripByteCounts was not set at all) the image is loaded correctly. I tried
```
InputStream adapter;
if (stripTileByteCounts != null && stripTileByteCounts[i] > 0) {
    adapter = createStreamAdapter(imageInput, stripTileByteCounts[i]);
} else {
    adapter = createStreamAdapter(imageInput);
}
```
at com/twelvemonkeys/imageio/plugins/tiff/TIFFImageReader.java:1066 and it worked well.
I am no TIFF expert so I do not know if this could have unwanted side effects.
And unfortunately I do not have a sample file I can attach. If needed I could ask our customer if they can produce an innocent sample file though.

Kind regards,
Stephan

**Comment by haraldk** [comment]
Hi Stephan,

Thank you for reporting! I think your suggestion makes sense, and will make the reader more robust. As you say, if libTiff ignores it, then we should probably too. 

However, I need a test case as well, so it would really help if your customer could produce a sample file for that purpose (small is good! 😊). If not, we could probably create/modify a sample file with this issue from a valid file. But I usually prefer having a "real world" example, so please ask them. 


Best regards,

-- 
Harald K

**Comment by stbstb** [comment]
So. Got an image. As it was produced by a multi-function device it was only possible to get a DIN-A4-image, sorry.
[CCITT-G4-300dpi-StripByteCounts0.zip](https://github.com/haraldk/TwelveMonkeys/files/2500885/CCITT-G4-300dpi-StripByteCounts0.zip)

**Comment by haraldk** [comment]
Excellent!

I guess that's fine, it's not too many bytes anyway...

I have a fix, that also emits a warning in the case of a strip byte count being zero, just in case. :-)

Will push to master soon.

-- 
Harald K

**Comment by haraldk** [comment]
Fixed in 7985cdd4a7c459cace5ee247426a968c37c8964f.
